internal/server/object_graph.go

```diff
@@ -0,0 +1,87 @@
+package server
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/satsuma-data/node-gateway/internal/checks"
+	"github.com/satsuma-data/node-gateway/internal/client"
+	"github.com/satsuma-data/node-gateway/internal/config"
+	"github.com/satsuma-data/node-gateway/internal/metadata"
+	"github.com/satsuma-data/node-gateway/internal/metrics"
+	"github.com/satsuma-data/node-gateway/internal/route"
+	"go.uber.org/zap"
+)
+
+type objectGraph struct {
```

Comment:
Yep, `objectGraph` is generic but it follows somewhat accepted Dependency Injection terminology: https://blog.ploeh.dk/2011/07/28/CompositionRoot/ . I initially tried more accurate names but it seemed like they 1) were pretty long, and 2) yet didn't seem to provide too much value. So at some point I switched to `ObjectGraph` (because, in DI terms, that is indeed the object graph for that app) and it all seemed much nicer. Especially because the type name is never actually used outside of this file.

Reply:
> Also, maybe moving RPCHandler in this file would help readability?

I think this sounds reasonable. I was a bit too fed up fighting with dependency wiring by the time I got this working and I didn't really want to do further refactoring of deps right now, tbh 😀

Reply:
sounds good!